pkg/clusterconditions/cache/cache.go

```diff
@@ -144,14 +143,14 @@ func (c *Cache) Match(ctx context.Context, condition *configv1.ClusterCondition)
 	if result, ok := c.MatchResults[key]; ok && !result.When.IsZero() {
 		return result.Match, result.Error
 	}
-	return false, errors.New("client-side throttling: this cached cluster condition request has been queued for later execution")
+	return false, errors.New("evaluation is throttled")
 }
 
 // expireStaleMatchResults removes entries from MatchResults if their
 // last-evaluation When is more than Expiration ago.  For MatchResults
 // entries which have never been evaluated, the last-request Access
 // time is used instead.
-func (c *Cache) expireStaleMatchResults(ctx context.Context, now time.Time) {
+func (c *Cache) expireStaleMatchResults(now time.Time) {
```

Comment:
I have included these cleanups in the same commit because they are local to the code change and not too invasive in the diff, unless there's a strong opposition to that I'd like to keep them

Reply:
I don't mind squashing close changes in, but dropping `ctx` blocks us from doing anything that might take time in this call.  I guess we could just update all the call sites again, if some future evolution adds some time-taking logic to this function?

Reply:
I agree with @wking that `dropping ctx blocks us from doing anything that might take time in this call`

Reply:
I'm in the YAGNI camp here :) We can easily wire the context back when and if it is needed. Both these functions are internal, single callsite helpers, having them consume `ctx` is just confusing for whoever reads the code now (did someone just forget to use it? did someone forget to remove it?).